KomaMRIFiles/src/Phantom/Phantom.jl

```diff
@@ -0,0 +1,178 @@
+        elseif type == "Indexed"
+            index = read(param["values"])
+            if Ns == length(index)
+                table = read(param["table"])
+                N = read_attribute(param, "N")
+                if N == length(table)
+                    values = table[index]
+                else
+                    print("Error: $(label) table dimensions mismatch")
+                end
+            else
+                print("Error: $(label) vector dimensions mismatch")
+            end
+        elseif type == "Default"
+            values = "Default"
+        end
+    else
+        values = read(param["values"])
+    end
+    return values
+end
+
+function import_motion(Ns::Int, motion_group::HDF5.Group)
+    model = read_attribute(motion_group, "model")
+    if model == "SimpleMotion"
+        types_group = motion_group["types"]
+        types = SimpleMotionType[]
+        for key in keys(types_group)
+            type_group = types_group[key]
+            type_str = split(key, "_")[2]
+            @assert type_str in last.(split.(string.(subtypes(SimpleMotionType)), ".")) "Simple Motion Type: $(type_str) has not been implemented in KomaMRIBase $(KomaMRIBase.__VERSION__)"
+            for SMT in subtypes(SimpleMotionType)
+                args = []
+                if type_str == last(split(string(SMT), "."))
+                    for key in fieldnames(SMT)
+                        push!(args, read_attribute(type_group, string(key)))
+                    end
+                    push!(types, SMT(args...))
+                end
+            end
+        end
+        return SimpleMotion(vcat(types...))
+    elseif model == "ArbitraryMotion"
+        dur = read(motion_group["duration"])
+        K = read_attribute(motion_group, "K")
+        dx = zeros(Ns, K - 1)
+        dy = zeros(Ns, K - 1)
+        dz = zeros(Ns, K - 1)
+        for key in HDF5.keys(motion_group)
+            if key != "duration"
+                values = read_param(motion_group[key])
+                if key == "dx"
+                    dx = values
+                elseif key == "dy"
+                    dy = values
+                elseif key == "dz"
+                    dz = values
+                end
+            end
+        end
+        return ArbitraryMotion(dur, dx, dy, dz)
+    end
+end
+
+
+
+"""
+write_phantom(ph,filename)
+
+Writes a (.phantom) file from a Phantom struct.
+"""
+# By the moment, only "Explicit" type 
+# is considered when writing .phantom files
+function write_phantom(obj::Phantom, filename::String)
+    # Create HDF5 phantom file
+    fid = h5open(filename, "w")
+
+    # Root attributes
+    HDF5.attributes(fid)["Version"] = "0.1"
+    HDF5.attributes(fid)["Name"] = obj.name
+    HDF5.attributes(fid)["Ns"] = length(obj.x)
+    dims = get_dims(obj)
+    HDF5.attributes(fid)["Dims"] = sum(dims)
+    fields = fieldnames(Phantom)[2:end]
+    # Spin initial positions
+    pos = create_group(fid, "position")
+    for i in 1:3
```

Comment:
The phantom should always have `x`, `y`, `z`. This will save only the ones that are not 0, so if `x=[0]` is not defined, when reading the `fields` variable will not have `x`, and the `Phantom` constructor will throw an error when reading the phantom.